apis/core/types_installation.go

```diff
@@ -40,6 +40,10 @@ type Installation struct {
 
 // InstallationSpec defines a component installation.
 type InstallationSpec struct {
+	// Context defines the current context of the installation.
+	// +optional
+	Context string `json:"context,omitempty"`
```

Comment:
Shouldn't that be a reference to a context resource, consisting of name and namespace?
If a context resource is namespaced but referenced only by name, the namespace would need to be defaulted or we have to ensure that there exists exactly one context with that name across all namespaces (which is strongly counter-intuitive regarding the concept of namespaces).

Reply:
OK, now that I think about it, it doesn't make any sense to allow referencing contexts in arbitrary namespaces, as this would grant users access to other tenants' registries.

Reply:
So the namespace of the context is always the same as the one where the installation is in? And there is an exception for the `default` context which lies in the `default` namespace?

Reply:
> So the namespace of the context is always the same as the one where the installation is in? And there is an exception for the default context which lies in the default namespace?

yes it is required that the context is in the same namespace as the installation.
Therefore no namespace is needed.

If no reference is given, the default context is used. There is not just one default context in the default namespace.
The concept is rather similar to the default service account. Every namespace has their own default context.